tests/phpunit/tests/blocks/getBlockTemplates.php

```diff
@@ -221,4 +221,51 @@ public function data_get_block_templates_should_respect_posttypes_property() {
 			),
 		);
 	}
+
+	/**
+	 * Test _get_block_templates_paths() for a valid theme dir.
+	 *
+	 * @ticket 58196
+	 * @covers ::_get_block_templates_paths
+	 */
+	public function test_get_block_templates_paths_dir_exists() {
+		$theme_dir = get_template_directory();
+		// templates in the current theme.
+		$templates = array(
+			'parts/small-header.html',
+			'templates/custom-single-post-template.html',
+			'templates/index.html',
+			'templates/page-home.html',
+			'templates/page.html',
+			'templates/single.html',
+		);
+
+		$expected_template_paths = array_map(
+			static function ( $template ) use ( $theme_dir ) {
+				return $theme_dir . '/' . $template;
+			},
+			$templates
+		);
+
+		$template_paths = _get_block_templates_paths( $theme_dir );
+		sort( $expected_template_paths );
+		sort( $template_paths );
+		$this->assertSame( $expected_template_paths, $template_paths );
```

Comment:
WP Unit tests have a helper for this.

```suggestion
		$this->assertSameSets( $expected_template_paths, $template_paths );
```

Reply:
Used `assertSameSets` but interestingly it runs 2 additional assertions internally.